tests/testcases.docopt

```diff
@@ -950,8 +950,35 @@ local options: --baz
 other options:
  --egg
  --spam
--not-an-option-
```

Comment:
Don't delete this line

Reply:
Ah, so the reason I got rid of this is that with docopt 0.6.2 this is actually a valid way to specify an option. In docopt-ng currently, option descriptions have to have leading whitespace, but not in 0.6.2.

I could put it back with a different name, like
```
-option-without-whitespace
```

although the "docopt 0.6.2 compatibility" test below also has options without leading whitespace.

Reply:
gotcha. I guess it follows the `line starting with - or --` rule, so you're right the other tests cover this case.

But, separate from the leading whitespace thing, I'm still hung up on the
-starts-with-single-dash-but-is-long, that isn't intuitive to me if that should be a long or short option.

Testing:
```
-long-solo
-s-d --single--double
--double-single -d-s
```
results in
`{"-long-solo": false, "--single--double": false, "--double-single": false}`
which I think makes sense, but probably should add a test case to make it official. Can add in follow up PR

Reply:
Oh, I missed that there was only one dash on that option. The option description parsing code is very lenient, it just assumes the doc is formatted according to the spec, and does almost no validation. The spec does have quite a clear definition of short vs long option in general (not specifically for option descriptions):

> Words starting with one or two dashes (with exception of "-", "--" by themselves) are interpreted as short (one-letter) or long options, respectively.
>
> * Short options can be "stacked" meaning that -abc is equivalent to -a -b -c.

And the usage pattern / argv parsers (the `Pattern` classes) do implement this behaviour (slight quirk with the hyphen):

```pycon
>>> docopt.docopt('usage: prog [-single-dash]', []) 
{'--': False, '-a': False, '-d': False, '-e': False, '-g': False, 
 '-h': False, '-i': False, '-l': False, '-n': False, '-s': 0}
```

And the argv parser/matcher doesn't recognise the the single dash option:
```pycon
>>> docopt.docopt('usage: prog [options]\noptions: -single-dash\n', '-single-dash')
usage: prog [options]
options: -single-dash
An exception has occurred, use %tb to see the full traceback.
```
I'd suggest following the behaviour implemented in the Pattern code, as it is significantly more detailed/careful than than the option default parsing code.